cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1215,24 +1215,60 @@ struct get_page_schema {
   __device__ size_type operator()(gpu::PageInfo const& page) { return page.src_col_schema; }
 };
 
+struct input_col_info {
+  int const schema_idx;
+  size_type const nesting_depth;
+};
+
+/**
+ * @brief Converts a 1-dimensional index into page, depth and column indices used in
+ * allocate_columns to compute columns sizes.
+ *
+ * The input index will iterate through pages, nesting depth and column indices in that order.
+ */
+struct reduction_indices {
+  const size_t _page_idx;
+  const size_type _depth_idx;
+  const size_type _col_idx;
```

Comment:
```suggestion
  size_t const _page_idx;
  size_type const _depth_idx;
  size_type const _col_idx;
```
"east const" is preferred in libcudf since there is only one rule to follow: what's left of `const` is constant